src/plugins/console/server/routes/api/console/proxy/create_handler.ts

```diff
@@ -116,7 +116,16 @@ export const createHandler =
   }: RouteDependencies): RequestHandler<unknown, Query, Body> =>
   async (ctx, request, response) => {
     const { body, query } = request;
-    const { path, method, withProductOrigin } = query;
+    const { method, withProductOrigin } = query;
+    const [p, q = ''] = query.path.split('?');
+    let path = query.path;
+
+    // if there is a '+' sign in query e.g. ?q=create_date:[2020-05-10T08:00:00.000+08:00 TO *]
+    // node url encodes it as a whitespace which results in a faulty request
+    // we need to replace '+' with '%2b' to encode it correctly
+    if (/\+/g.test(q)) {
```

Comment:
I'm not a big fan doing it. At least it will be great to replace it only for exactly that case - I'm talking about creating a regexp for timestamp to don't affect other cases with +

Also,  from my perspective you fixed not a root cause and this kind of issue possible not only for +.

Reply:
Ok... I've debugged that a little. What do you think about moving that code into `create_handler.ts:toURL ` method?